src/challenges/challengeOne/5.Modal/Modal.test.tsx

```diff
@@ -1,8 +1,54 @@
-// import React from "react";
-// import { render, screen } from "@testing-library/react";
-// import userEvent from "@testing-library/user-event";
-// import { Modal } from "./Modal";
-
-// describe("Modal", () => {
-//   it("renders...", () => {});
-// });
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Modal, ModalProps } from "./Modal";
+
+describe("Modal", () => {
+  const onClose = jest.fn();
+  const onConfirm = jest.fn();
+
+  const renderComponent = (additionalProps?: Partial<ModalProps>) => {
+    const defaultProps: ModalProps = {
+      isOpen: true,
+      onClose,
+      onConfirm,
+      children: <div>Modal Content</div>,
+      headerText: "Modal Header",
+      showActions: true,
+    };
+
+    render(<Modal {...defaultProps} {...additionalProps} />);
+  };
+
+  it("renders without error", () => {
+    renderComponent();
+    expect(screen.getByText("Modal Header")).toBeInTheDocument();
+  });
+
+  it("renders children content", () => {
+    renderComponent();
+    expect(screen.getByText("Modal Content")).toBeInTheDocument();
+  });
+
+  it("closes the modal when close button is clicked", async () => {
+    renderComponent();
+    await userEvent.click(screen.getByRole("button", { name: /cancel/i }));
+    expect(onClose).toHaveBeenCalledTimes(1);
```

Comment:
This is also an option

```suggestion
    expect(onClose).toHaveBeenCalled();
```

Reply:
Great. I'll save these inputs for the best.